### PR TITLE
Foxy Branch Updates

### PR DESCRIPTION
**Pull request by vikrammeyer**
- Removed gitsubmodule and replaced with cmake macro to download the CTRE libs
- Added launch file for multiple talon nodes since components did not work due to the multiple diagnostic servers issue

### CODE REVIEW

**Comment by nkalupahana** [comment]
What's the purpose of the `foxy` branch, exactly?

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +12,63 @@ else()
   message(FATAL_ERROR "Unsupported architecture for CTRE Phoenix")
 endif()
 
+set(CTRE_VERSION "5.18.3")
+
+#Macro for downloading CTRE Libraries
```
Comment (nkalupahana) [inline CMakeLists.txt]:
```suggestion
# Macro for downloading CTRE Libraries
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -63,16 +110,14 @@ set(PHOENIX_LIBRARIES CTRE_Phoenix
...
-  add_library(${LIBRARY} STATIC IMPORTED)
-  set(LIBRARY_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/Phoenix-Linux-SocketCAN-Example/lib/${CTRE_ARCH}/lib${LIBRARY}.a)
-  SET_TARGET_PROPERTIES(${LIBRARY} PROPERTIES IMPORTED_LOCATION ${LIBRARY_LOCATION})
-  install (
-    FILES ${LIBRARY_LOCATION}
-    DESTINATION lib
-  )
-endforeach()
+
+# Download the CTRE libraries
```
Comment (nkalupahana) [inline CMakeLists.txt:114]: Are you sure the libs should be tabbed out like this here? This looks wack.

File: include/ros_phoenix/base_component.hpp [hunk 3]
```diff
@@ -212,7 +212,7 @@ namespace ros_phoenix
 
         void onTimer()
         {
-            ctre::phoenix::unmanaged::FeedEnable(this->watchdog_ms_);
+            ctre::phoenix::unmanaged::Unmanaged::FeedEnable(this->watchdog_ms_);
```
Comment (nkalupahana) [inline include/ros_phoenix/base_component.hpp:216]: Can we get a link to the changelog here with the version that this changed? Will be useful for reference

File: launch/demo-components.launch.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import launch
```
Comment (nkalupahana) [inline launch/demo-components.launch.py:1]: This file doesn't work, does it? If it doesn't, I don't think it should be committed in.

File: launch/demo-nodes.launch.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+from launch import LaunchDescription
+from launch_ros.actions import Node
+
+
+def generate_launch_description():
+    """Generate launch description with multiple nodes."""
```
Comment (nkalupahana) [inline launch/demo-nodes.launch.py]:
```suggestion
    """Launches talons for basic 4-wheel drive."""
```

File: src/phoenix_container.cpp [hunk 6]
```diff
@@ -11,7 +11,7 @@ int main(int argc, char *argv[])
 {
     // Start Phoenix diagnostic server in background
```
Comment (nkalupahana) [inline src/phoenix_container.cpp]:
```suggestion
    // Stops the launch of the Phoenix diagnostic server so only one instance is started
```

Comment (nkalupahana) [reply src/phoenix_container.cpp]: This is a a sample comment, I'm not entirely sure if this is right

**Review by nkalupahana — CHANGES_REQUESTED** [review]
Almost there; see comments.

File: launch/demo-components.launch.py (hunk 4 above)
Comment (vikrammeyer) [reply launch/demo-components.launch.py:1]: I was just gonna leave this cause its more of joey's work with components if he wants to fix that at a later date

Comment (nkalupahana) [reply launch/demo-components.launch.py:1]: Ah, okay. How about we add a comment to the top that says, like:
```
"""
DO NOT LAUNCH THIS FILE!

Due to CTRE diagnostic server issues, running multiple talons in the same process
(as is done through ComposableNode) causes all but one talon node to fail. Thus,
this launch file does not work. For now, please use `demo-nodes.launch.py` instead.
"""
```

File: CMakeLists.txt (hunk 2 above)
Comment (vikrammeyer) [reply CMakeLists.txt:114]: yeah that's how joey has it in the original macro on master

Comment (nkalupahana) [reply CMakeLists.txt:114]: it's yucky thoughhh 😂 your call, do what you think is best

**Review by nkalupahana — APPROVED** [review]

**Pull request merged**